webapp/channels/src/components/advanced_text_editor/send_button/send_post_options/core_menu_options.tsx

```diff
@@ -23,6 +30,69 @@ function CoreMenuOptions({handleOnSelect, channelId}: Props) {
         isDM,
     } = useTimePostBoxIndicator(channelId);
 
+    const recentlyUsedCustomDate = useSelector((state: GlobalState) => getPreference(state, scheduledPosts.SCHEDULED_POSTS, scheduledPosts.RECENTLY_USED_CUSTOM_TIME));
+
+    interface RecentlyUsedCustomDate {
+        update_at?: number;
+        timestamp?: number;
+    }
+
+    let recentlyUsedCustomDateVal: RecentlyUsedCustomDate = {};
+
+    if (recentlyUsedCustomDate) {
+        try {
+            recentlyUsedCustomDateVal = JSON.parse(recentlyUsedCustomDate) as RecentlyUsedCustomDate;
+        } catch (e) {
+            recentlyUsedCustomDateVal = {};
+        }
+    }
+
+    function isTimestampWithinLast30Days(timestamp: number, timeZone = 'UTC') {
+        if (!timestamp || isNaN(timestamp)) {
+            return false;
+        }
+        const usedDate = DateTime.fromMillis(timestamp).setZone(timeZone);
+        const now = DateTime.now().setZone(timeZone);
+        const thirtyDaysAgo = now.minus({days: 30});
+
+        return usedDate >= thirtyDaysAgo && usedDate <= now;
+    }
+
+    let recentCustomTime = null;
+    const handleRecentlyUsedCustomTime = useCallback((e) => handleOnSelect(e, recentlyUsedCustomDateVal.timestamp!), [handleOnSelect, recentlyUsedCustomDateVal.timestamp]);
+    if (
+        recentlyUsedCustomDateVal &&
+        typeof recentlyUsedCustomDateVal.update_at === 'number' &&
+        typeof recentlyUsedCustomDateVal.timestamp === 'number' &&
+        isTimestampWithinLast30Days(recentlyUsedCustomDateVal.update_at, userCurrentTimezone)
+    ) {
+        const timestamp = (
+            <Timestamp
+                value={recentlyUsedCustomDateVal.timestamp}
+                timeZone={userCurrentTimezone}
+                useRelative={false}
+                useDate={{weekday: 'long'}}
+                useTime={{hour: 'numeric', minute: 'numeric'}}
+            />
+        );
+
+        recentCustomTime = [
```

Comment:
Why define it as a list instead of just as a JSX component inside a fragment?

The final result is practically the same, and looks more idiomatic. But 0/5.

Reply:
the way it ended up being defined as a fragment, was requiring to have a key in it so it could be tracked correctly by React. So, this ended up working fine and without the need of fragments involved.